diff --git a/webprofiler/src/DataCollector/FormsDataCollector.php b/webprofiler/src/DataCollector/FormsDataCollector.php
index 79f6332..a22c870 100644
--- a/webprofiler/src/DataCollector/FormsDataCollector.php
+++ b/webprofiler/src/DataCollector/FormsDataCollector.php
@@ -41,7 +41,7 @@ class FormsDataCollector extends DataCollector implements DrupalDataCollectorInt
    * @return array
    */
   public function getForms() {
-    return $this->data['forms'];
+    return (!empty($this->data['forms']) && is_array($this->data['forms'])) ? $this->data['forms'] : [];
   }
 
   /**
